### PR TITLE
Make most interactions with the db panic on failure instead of returning Result

### PR DESCRIPTION
**Pull request by mystenmark**

### CODE REVIEW

File: crates/sui-core/src/authority/authority_store_tables.rs [hunk 1]
```diff
@@ -492,42 +492,47 @@ impl AuthorityPerpetualTables {
...
     }
-}
 
-impl ObjectStore for AuthorityPerpetualTables {
-    /// Read an object and return it, or Ok(None) if the object was not found.
-    fn get_object(
-        &self,
-        object_id: &ObjectID,
-    ) -> Result<Option<Object>, sui_types::storage::error::Error> {
+    // fallible get object methods for sui-tool, which may need to attempt to read a corrupted database
```
Comment (mystenmark) [inline crates/sui-core/src/authority/authority_store_tables.rs:496]: 👀

File: crates/sui-core/src/authority.rs [hunk 2]
```diff
@@ -1281,10 +1281,13 @@ impl AuthorityState {
...
     ) -> SuiResult<TransactionEffects> {
-        self.get_transaction_cache_reader()
+        // XXx
+        Ok(self
+            .get_transaction_cache_reader()
             .notify_read_executed_effects(&[digest])
             .await
-            .map(|mut r| r.pop().expect("must return correct number of effects"))
+            .pop()
+            .expect("must return correct number of effects"))
```
Comment (bmwill) [inline crates/sui-core/src/authority.rs:1290]: Is the 'xxx' here to remind you to do something here?

File: crates/sui-core/src/authority.rs [hunk 3]
```diff
@@ -4175,10 +4172,11 @@ impl AuthorityState {
     pub fn get_transaction_output(
         &self,
         transaction_digest: &TransactionDigest,
+        // XXX cleanup
```
Comment (bmwill) [inline crates/sui-core/src/authority.rs:4175]: Ah i see there are other cleanups you could do

File: crates/sui-core/src/authority/authority_store_tables.rs (hunk 1 above)
Comment (bmwill) [reply crates/sui-core/src/authority/authority_store_tables.rs:496]: lgtm

**Review by bmwill — APPROVED** [review]

**Pull request merged**